src/VisualStudio/Core/Def/Implementation/Diagnostics/VisualStudioWorkspaceDiagnosticAnalyzerProviderService.cs

```diff
@@ -34,15 +35,18 @@ internal partial class VisualStudioWorkspaceDiagnosticAnalyzerProviderService :
         private static readonly AnalyzerAssemblyLoader s_analyzerAssemblyLoader = new AnalyzerAssemblyLoader();
 
         [ImportingConstructor]
-        public VisualStudioWorkspaceDiagnosticAnalyzerProviderService(VisualStudioWorkspaceImpl workspace)
+        public VisualStudioWorkspaceDiagnosticAnalyzerProviderService(
+            [Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider)
         {
+            var dte = (EnvDTE.DTE)serviceProvider.GetService(typeof(EnvDTE.DTE));
```

Comment:
The actual complexity of what does and doesn't need to get moved is tricky. Since dependencies could sneak in all sorts of places, I have dotnet/roslyn-internal#1380 to have a test to catch future mistakes.